datalad/support/gitrepo.py

```diff
@@ -671,21 +674,174 @@ def git_remote_update(self, name='', verbose=False):
         self._git_custom_command('', 'git remote %s update %s' % (name, v),
                                  expect_stderr=True)
 
-    def git_fetch(self, name, options=''):
+    # TODO: centralize all the c&p code in fetch, pull, push
+    # TODO: document **kwargs passed to gitpython
+    def fetch(self, remote=None, refspec=None, progress=None, all=False, **kwargs):
+        # TODO: options=> **kwargs):
+        """
+        """
+        # Note: Apparently there is no explicit (fetch --all) in gitpython,
+        #       but fetch is always bound to a certain remote instead.
+        #       Therefore implement it on our own:
+        if remote is None:
+            if refspec is not None:
+                # conflicts with using tracking branch or fetch all remotes
+                # For now: Just fail.
+                # TODO: May be check whether it fits to tracking branch
+                raise ValueError("refspec specified without a remote. (%s)" %
+                                  refspec)
+            if all:
+                remotes_to_fetch = self.repo.remotes
+            else:
+                # No explicit remote to fetch.
+                # => get tracking branch:
+                tb = self.repo.active_branch.tracking_branch().name
+                if tb:
+                    parts = tb.split('/')
+                    assert len(parts) == 2
+                    remotes_to_fetch = [self.repo.remote(parts[0])]
+                    refspec = parts[1]
+                else:
+                    # No remote, no tracking branch
+                    # => fail
+                    raise ValueError("No remote specified to fetch from nor a "
+                                     "tracking branch is set up.")
+        else:
+            remotes_to_fetch = [self.repo.remote(remote)]
+
+        for rm in remotes_to_fetch:
+            fetch_url = \
+                rm.config_reader.get('fetchurl'
+                                     if rm.config_reader.has_option('fetchurl')
+                                     else 'url')
+            if fetch_url.startswith('ssh:'):
```

Comment:
I agree. Given that we had several discussions about URL handling recently, I'll make it an issue on its own.
Therefore: #493